packages/llama/src/Llama.ts

```diff
@@ -0,0 +1,43 @@
+import { type VoteActionParams, compressJson } from '@rabbitholegg/questdk'
+import { LLAMA_ABI_CAST_APPROVE, LLAMA_ABI_CAST_DISAPPROVE } from './abi.js'
+import { CHAIN_ID_ARRAY } from './chain-ids.js'
+import { LLAMA_TOKENS } from './token-addresses.js'
+
+export const vote = async (voteParams: VoteActionParams) => {
+  // This is the information we'll use to compose the Transaction object
+  const { chainId, project, proposalId, support } = voteParams
+
+  const supportIsBoolean = support !== undefined && typeof support === 'boolean'
+
+  let abi = []
+  if (supportIsBoolean) {
+    // if the support param is a boolean, we only want to use the abi that matches the boolean
+    if (support) {
+      abi = LLAMA_ABI_CAST_APPROVE
+    } else {
+      abi = LLAMA_ABI_CAST_DISAPPROVE
+    }
+  } else {
+    // if the support param is a filter or undefined, then we can match either an approve or disapprove
+    abi = LLAMA_ABI_CAST_APPROVE.concat(LLAMA_ABI_CAST_DISAPPROVE)
+  }
+
+  return compressJson({
+    chainId: chainId, // The chainId of the source chain
+    to: project, // The contract address of the governance platform
+    input: {
+      $abi: abi,
+      ...(proposalId !== undefined && { actionInfo: { id: proposalId } }),
+    },
+  })
+}
+
+export const getSupportedTokenAddresses = async (_chainId: number) => {
+  // Given a specific chain we would expect this function to return a list of supported token addresses
+  return LLAMA_TOKENS[_chainId]
```

Comment:
Im not sure if we need to return token addresses for a vote plugin 🤔. @Quazia wdyt? 

For tokens to work correctly here, the tokens need to be indexable on coingecko services, so we can obtain the icon and decimals.

https://github.com/rabbitholegg/rh/blob/main/src/services/internal/v1/tasks.ts#L11-L40

Reply:
I ended up just setting all the factory contract addresses which won't work on coingecko eiterh 🙃  not sure if it's better to return an empty array?

Reply:
Yeah probably best to leave it empty. The only param we could map to the supported token is the project address and that's probably best off as just an address entry vs a drop-down.